src/vscode-bicep/src/azure/types.ts

```diff
@@ -14,3 +15,42 @@ export type AzureLoginStatus =
   | "LoggingIn"
   | "LoggedIn"
   | "LoggedOut";
+
+type DeploymentScopeBase = {
+  portalUrl: string;
+  tenantId: string;
+};
```

Comment:
nit: One trick to simplify `DeploymentScope` is to make `DeploymentScopeBase` a utility type:

```typescript
type DeploymentScopeBase<T> = {
  portalUrl: string;
  tenantId: string;
} & T;

export type DeploymentScope = DeploymentScopeBase<
  | {
      scopeType: "resourceGroup";
      subscriptionId: string;
      resourceGroup: string;
    }
  | {
      scopeType: "subscription";
      location: string;
      subscriptionId: string;
    }
  | {
      scopeType: "managementGroup";
      associatedSubscriptionId: string;
      location: string;
      managementGroup: string;
    }
  | {
      scopeType: "tenant";
      associatedSubscriptionId: string;
      location: string;
    }
>;
```

Reply:
Thanks! I felt like there was something I could improve here, but didn't know what.